static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -29,22 +29,51 @@ define([
                     if (response && response.subscriptions && response.subscriptions.length) {
                         isSubscribed = true;
                     }
-                    $(buttonEl).removeClass('is-updating is-updating-subscriptions');
                     updateButton(buttonEl, isSubscribed);
                 }
             });
         });
     }
 
-    function enhanceEmailPreferences() {
-        $.forEachElement('.email-subscription__button', reqwestEmailSubscriptionUpdate);
-        $.forEachElement('.save__button', reqwestEmailSubscriptionUpdate);
+    function confirmUnsubscriptionFromAll(buttonEl) {
+        fastdom.write(function () {
+            $(buttonEl).addClass('email-unsubscribe--confirm js-confirm-unsubscribe');
+            $('.email-unsubscribe-all__label').toggleClass('hide');
+        });
     }
 
-    function encodeFormData(csrfToken, buttonVal, htmlPreference) {
-        return 'csrfToken=' + encodeURIComponent(csrfToken) + '&' +
-        'addEmailSubscription=' + encodeURIComponent(buttonVal) + '&' +
-        'htmlPreference=' + encodeURIComponent(htmlPreference);
+    function unsubscribeFromAll(buttonEl) {
+        bean.on(buttonEl, 'click', function () {
+            if ($(buttonEl).hasClass('js-confirm-unsubscribe')) {
+                addUpdatingState(buttonEl);
+                resetUnsubscribeFromAll(buttonEl);
+                var subscribedButtons = getAllSubscribedButtons();
+                for (var i = 0; i < subscribedButtons.length; i++) {
+                    subscribedButtons[i].click();
```

Comment:
as we discussed, I'm a bit wary of firing off loads of requests when we could actually batch them into one. I'll tweak the backend so it can accept multiple add/remove requests in one